instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/servlet/ServerSpanNaming.java

```diff
@@ -50,7 +50,7 @@ private ServerSpanNaming(Source initialSource) {
   public static void updateServerSpanName(
       Context context, Source source, Supplier<String> serverSpanName) {
     Span serverSpan = ServerSpan.fromContextOrNull(context);
-    if (serverSpan == null) {
+    if (serverSpan == null || !serverSpan.isRecording()) {
```

Comment:
Nit: can you add a comment saying that `serverSpan.isRecording()` is just an optimization?